tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/factoriestransformers/PrincipalTransformersTest.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.hal.testsuite.test.configuration.elytron.factoriestransformers;
+
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.dmr.ModelNode;
+import org.jboss.hal.testsuite.fragment.AddResourceDialogFragment;
+import org.jboss.hal.testsuite.fragment.FormFragment;
+import org.jboss.hal.testsuite.fragment.TableFragment;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.jboss.hal.dmr.ModelDescriptionConstants.NAME;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.PATTERN;
+import static org.jboss.hal.testsuite.test.configuration.elytron.ElytronFixtures.*;
+
+@RunWith(Arquillian.class)
+public class PrincipalTransformersTest extends AbstractFactoriesTransformersTest {
```

Comment:
I would personally split this test class into corresponding principal transformers e.g: AggregatePrincipalTransformerTest, RegexPrincipalTransformerTest etc... This way it would give us more flexibility to expand the tests, e.g create principal transformer with invalid data types inserted into the dialog